src/rsa/signing.rs

```diff
@@ -251,13 +251,20 @@ impl RSAKeyPair {
                 let n = n.into_modulus::<N>()?;
                 let oneRR_mod_n = bigint::One::newRR(&n)?;
 
+                // Swap p and q if p < q.
+                let (p, q, dP, dQ, pq_swapped) = match q.verify_less_than(&p) {
+                    Ok(_)  => (p, q, dP, dQ, false),
+                    Err(_) => (q, p, dQ, dP, true),
+                };
```

Comment:
Please also calculate the new `qInv` here instead of doing it below. It would be clearer. Or, if there's a reason not to calculate it here, add a comment about why the new `qInv` isn't calculated here.